lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -75,59 +75,80 @@ class AwsCompileFunctions {
     }
   }
 
-  downloadPackageArtifact(functionName) {
+  async downloadPackageArtifact(functionName) {
     const { region } = this.options;
     const S3 = new AWS.S3({ region });
 
     const functionObject = this.serverless.service.getFunction(functionName);
+    if (functionObject.image) return;
+
     const artifactFilePath =
       _.get(functionObject, 'package.artifact') ||
       _.get(this, 'serverless.service.package.artifact');
 
     const regex = new RegExp('s3\\.amazonaws\\.com/(.+)/(.+)');
     const match = artifactFilePath.match(regex);
 
-    if (match) {
-      return new BbPromise((resolve, reject) => {
-        const tmpDir = this.serverless.utils.getTmpDirPath();
-        const filePath = path.join(tmpDir, match[2]);
-
-        const readStream = S3.getObject({
-          Bucket: match[1],
-          Key: match[2],
-        }).createReadStream();
-
-        const writeStream = fs.createWriteStream(filePath);
-
-        readStream.on('error', error => reject(error));
-        readStream
-          .pipe(writeStream)
-          .on('error', reject)
-          .on('close', () => {
-            if (functionObject.package.artifact) {
-              functionObject.package.artifact = filePath;
-            } else {
-              this.serverless.service.package.artifact = filePath;
-            }
-            return resolve(filePath);
-          });
-      });
+    if (!match) return;
+    await new Promise((resolve, reject) => {
+      const tmpDir = this.serverless.utils.getTmpDirPath();
+      const filePath = path.join(tmpDir, match[2]);
+
+      const readStream = S3.getObject({
+        Bucket: match[1],
+        Key: match[2],
+      }).createReadStream();
+
+      const writeStream = fs.createWriteStream(filePath);
+      readStream
+        .pipe(writeStream)
+        .on('error', reject)
+        .on('close', () => {
+          if (functionObject.package.artifact) {
+            functionObject.package.artifact = filePath;
+          } else {
+            this.serverless.service.package.artifact = filePath;
+          }
+          return resolve(filePath);
+        });
+    });
+  }
+
+  async compileFunction(functionName) {
+    const cfTemplate = this.serverless.service.provider.compiledCloudFormationTemplate;
+    const functionResource = this.cfLambdaFunctionTemplate();
+    const functionObject = this.serverless.service.getFunction(functionName);
+    functionObject.package = functionObject.package || {};
+
+    if (!functionObject.handler && !functionObject.image) {
+      const errorMessage = [
+        `Either "handler" or "image" property needs to be set on function "${functionName}".`,
+        ' Please check the docs for more info',
```

Comment:
Do we have a link to the docs which we could add here and on L133?

Reply:
I'll actually remove this second line, I've left it as it was in case of validation for `handler` property, but it's totally superfluous as in case of crash link to general docs is always provided -> https://github.com/serverless/serverless/blob/c5ae9798d2feca03cbcf2290661a08442c2f1c7d/lib/classes/Error.js#L95-L96

Reply:
Sounds great! 👍

Reply:
Updated!